xbmc/addons/interfaces/GUI/AddonCallbacksGUI.cpp

```diff
@@ -1549,20 +1547,12 @@ void CAddonCallbacksGUI::ListItem_SetLabel2(void *addonData, GUIHANDLE handle, c
 
 void CAddonCallbacksGUI::ListItem_SetIconImage(void *addonData, GUIHANDLE handle, const char *image)
 {
-  CAddonInterfaces* helper = (CAddonInterfaces*) addonData;
-  if (!helper || !handle)
-    return;
-
-  static_cast<CFileItem*>(handle)->SetIconImage(image);
+  CLog::Log(LOGWARNING, "SetIconImage results in NOP and will soon be removed. Use setArt instead.");
 }
 
 void CAddonCallbacksGUI::ListItem_SetThumbnailImage(void *addonData, GUIHANDLE handle, const char *image)
 {
-  CAddonInterfaces* helper = (CAddonInterfaces*) addonData;
-  if (!helper || !handle)
-    return;
-
-  static_cast<CFileItem*>(handle)->SetArt("thumb", image);
+  CLog::Log(LOGWARNING, "SetThumbnailImage results in NOP and will soon be removed. Use setArt instead.");
```

Comment:
The `ListItem_SetIconImage` and `ListItem_SetThumbnailImage` if not called from header can be removed.